compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -128,6 +128,8 @@ static void addTileAndDistributePasses(
     OpPassManager &pm, bool useFuseTensorPadWithConsumerPass = true) {
   pm.addPass(createTileAndDistributeToWorkgroupsPass());
   auto &nestedModulePM = pm.nest<ModuleOp>();
+  nestedModulePM.addNestedPass<func::FuncOp>(
```

Comment:
Nice! I have a suggestion here that is not necessary for this op.

It seems like we can combine the "DecomposeAttentionPass" and "DecomposeSoftmaxPass" into a single pass with a new interface. A simple interface that just adds a single method, i.e. "decomposeOperation" that decomposes the operation into the required sequence. Then the pass is just walking over the IR and decomposing the ops as needed?

WDYT?

Also FYI @ftynse (since Alex seems to be interested)

Reply:
Sounds great! Where would I add such an interface?

Reply:
+1 for the interface. IMO, it can go to LinalgExtInterfaces.h/cpp. This sounds specific to Linalg at this point, at the ops it applies to are in LianlgExt.